core/source_generator/source_generator.cc

```diff
@@ -71,23 +71,35 @@ string prettySigForMethod(const core::GlobalState &gs, core::MethodRef method, c
     if (!flags.empty()) {
         flagString = fmt::format("{}.", fmt::join(flags, "."));
     }
+    string typeParamString = "";
+    if (!sym->typeArguments().empty()) {
+        typeParamString =
+            fmt::format("type_parameters({}).", fmt::map_join(sym->typeArguments(), ", ", [&](const auto &typeParam) {
+                            return typeParam.data(gs)->name.showAsSymbolLiteral(gs);
+                        }));
+    }
     string paramsString = "";
     if (!typeAndArgNames.empty()) {
         paramsString = fmt::format("params({}).", fmt::join(typeAndArgNames, ", "));
     }
 
-    auto oneline = fmt::format("{} {{ {}{}{} }}", sigCall, flagString, paramsString, methodReturnType);
-    if (oneline.size() <= MAX_PRETTY_WIDTH && typeAndArgNames.size() <= MAX_PRETTY_SIG_ARGS) {
-        return oneline;
+    auto oneline = sigCall.size() + flagString.size() + paramsString.size() + methodReturnType.size();
+    if (oneline <= MAX_PRETTY_WIDTH && typeAndArgNames.size() <= MAX_PRETTY_SIG_ARGS) {
+        return fmt::format("{} {{ {}{}{}{} }}", sigCall, flagString, typeParamString, paramsString, methodReturnType);
+    }
```

Comment:
technically I think this new calculation is off by 5 (because there were 3 spaces and 2 brace characters in the old calculation) but if that didn't cause any tests to fail I'm kind of fine just ignoring that. up to you

Reply:
I think for the sake of avoiding any potential future edge cases, it wouldn't hurt to include it.